src/cartesian/getTicks.ts

```diff
@@ -1,10 +1,47 @@
 import _ from 'lodash';
-import { CartesianTickItem } from '../../util/types';
-import { mathSign, isNumber } from '../../util/DataUtils';
-import { getStringSize } from '../../util/DOMUtils';
-import { Props as CartesianAxisProps } from '../CartesianAxis';
-import { Global } from '../../util/Global';
-import { getNumberIntervalTicks } from './utils';
+import { CartesianTickItem } from '../util/types';
+import { mathSign, isNumber } from '../util/DataUtils';
+import { getStringSize } from '../util/DOMUtils';
+import { Props as CartesianAxisProps } from './CartesianAxis';
+import { Global } from '../util/Global';
+import { getEveryNth } from '../util/getEveryNth';
+
+/**
+ * Given an array of ticks, find N, the lowest possible number for which every
+ * nTH tick in the ticks array isShow == true and return the array of every nTh tick.
+ * @param {CartesianTickItem[]} ticks An array of CartesianTickItem with the
+ * information whether they can be shown without overlapping with their neighbour isShow.
+ * @returns {CartesianTickItem[]} Every nTh tick in an array.
+ */
+export function getEveryNThTick(ticks: CartesianTickItem[]) {
+  let N = 1;
+  let previous: CartesianTickItem[] = ticks;
+  while (N < ticks.length) {
```

Comment:
Isn't this loop simpler with something like `getEveryNthWithCondition`?

```typescript
let N = 1;
let previous = getEveryNthWithCondition(ticks, N, (tickItem) => tickItem.isShow));
while (previous.length === Math.ceil(ticks.length / N)) {
  N++;
  previous = getEveryNthWithCondition(ticks, N, (tickItem) => tickItem.isShow));
}
return previous;
```

Reply:
It certainly is, good point.